game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/CombatValue.java

```diff
@@ -15,8 +15,14 @@
 
   boolean isDefending();
 
+  Collection<TerritoryEffect> getTerritoryEffects();
+
   GameData getGameData();
 
+  Collection<Unit> getFriendUnits();
```

Comment:
`getFriendUnits` and `getEnemyUnits` is used solely to send the friend and enemy units across the wire in `Player#selectCasualties`.  And the only implementation of `Player#selectCasualties` that uses the enemy and friend units is `AbstractProAi#selectCasualties`.  I've thought about putting a `@ChangeOnNextRelease` tag here, but I'm not exactly sure if there is something that could be done to remove the need for these two methods.